File: engine/engine-model/src/main/scala/tools/packet/buff/GiveBuff.scala
package tools.packet.buff

import client.MapleBuffStat
import net.opcodes.SendOpcode
import tools.Pair
import tools.packet.PacketInput

class GiveBuff(private var _buffId: Int, private var _buffLength: Int, private var _statIncreases: java.util.List[Pair[MapleBuffStat, java.lang.Integer]]) extends PacketInput {
  def buffId: Int = _buffId

  def buffLength: Int = _buffLength

  def statIncreases: java.util.List[Pair[MapleBuffStat, java.lang.Integer]] = _statIncreases

  override def opcode(): SendOpcode = SendOpcode.GIVE_BUFF
}

File: engine/engine-model/src/main/java/tools/packet/buff/GiveBuff.java
package tools.packet.buff;

import java.util.List;

import client.MapleBuffStat;
import net.opcodes.SendOpcode;
import tools.Pair;
import tools.packet.PacketInput;

public record GiveBuff(Integer buffId, Integer buffLength,
                       List<Pair<MapleBuffStat, Integer>> statIncreases) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GIVE_BUFF;
   }
}
